deltametrics/mask.py

```diff
@@ -48,51 +53,308 @@ def show(self, **kwargs):
     And visualize the mask:
-        >>> cmsk.show_mask()
+        >>> cmsk.show()
 
     .. plot:: mask/channelmask.py
 
     """
 
-    def __init__(self, arr, is_mask=False):
+    def __init__(self, velocity, topo, **kwargs):
         """Initialize the ChannelMask.
 
-        Intializing the channel mask requires an array of data, should be
-        two-dimensional.
+        Intializing the channel mask requires a flow velocity field and an
+        array of the delta topography.
 
         Parameters
         ----------
-        arr : ndarray
-            The data array to make the mask from.
+        velocity : ndarray
+            The velocity array to be used for mask creation.
+
+        topo : ndarray
+            The model topography to be used for mask creation.
+
+        Other Parameters
+        ----------------
+        velocity_threshold : float, optional
+            Threshold velocity above which flow is considered 'channelized'.
+            The default value is 0.3 m/s based on DeltaRCM default parameters
+            for sediment transport.
+
+        landmask : :obj:`LandMask`, optional
+            A :obj:`LandMask` object with a defined binary land mask.
+            If given, it will be used to help define the channel mask.
+
+        wetmask : :obj:`WetMask`, optional
+            A :obj:`WetMask` object with a defined binary wet mask.
+            If given, the landmask attribute it contains will be used to
+            determine the channel mask.
 
-        routine : str
-            Which routine to use to extract the mask.
+        topo_threshold : float, optional
+            Threshold depth to use for the OAM. Default is -0.5.
+
+        angle_threshold : int, optional
+            Threshold opening angle used in the OAM. Default is 75 degrees.
+
+        numviews : int, optional
+            Defines the number of times to 'look' for the OAM. Default is 3.
 
         is_mask : bool, optional
             Whether the data in :obj:`arr` is already a binary mask. For
             example, this should be set to True, if you have already binarized
             the data yourself, using custom routines, and want to just store
-            the data in the ChannelMask object.
-        """
-        super().__init__(mask_type='channel', data=arr)
+            the data in the LandMask object.
 
-        self.other_atts = 10
+        """
+        super().__init__(mask_type='channel', data=topo)
+        self.velocity = velocity
+
+        # assign **kwargs to self
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        # set default values for **kwargs if they do not exist
+        self.velocity_threshold = getattr(self, 'velocity_threshold', 0.3)
+        self.topo_threshold = getattr(self, 'topo_threshold', -0.5)
+        self.angle_threshold = getattr(self, 'angle_threshold', 75)
+        self.numviews = getattr(self, 'numviews', 3)
+        self.is_mask = getattr(self, 'is_mask', False)
```

Comment:
was there a reason you chose this approach rather than `pop`ping, e.g., `self.velocity_threshold = kwargs.pop('velocity_threshold', 0.3)`? Moreover, I think `kwargs` is best used when you will pass the high-level function input `kwargs` down several chains of lower-level functions, but you don't want to have to specify all possible options along the way. It's also useful for interacting with external libraries like `matplotlib` so we don't have to explicitly handle all available options there.

In this case here, I think these are mostly actually parameters relevant for the OAM methods. So, do you really need to set them in each `xMask` or can you just look for them within the `OAM` method? This could help reduce some of the repetitive nature here. 

Similarly, do you need to use `self.is_mask` or can `is_mask=False` just be an optional parameter in the `__init__` function signature and then where you need it below on Line 320 just use it as `if is_mask:`?